kernel/src/syscall/sigaltstack.rs

```diff
@@ -36,14 +36,20 @@ fn get_old_stack(
         return Ok(());
     }
 
-    let Some(old_stack) = old_stack else {
+    if let Some(old_stack) = old_stack {
+        debug!("old stack = {:?}", old_stack);
+
+        let stack = stack_t::from(old_stack.clone());
+        ctx.get_user_space().write_val(old_sig_stack_addr, &stack)?;
+    } else {
+        let old_stack = SigStack::new(0, SigStackFlags::SS_DISABLE, 0);
+        let stack = stack_t::from(old_stack);
```

Comment:
There's no need to create `SigStack` first, then convert it as `stack_t`. You can directly create a `stack_t`, then the conversion can be avoided.